### PR TITLE
feat: prepare force flag

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR adds force flag to prepare command, this flag allows reindexing of existing provider

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/prepare.ts [hunk 1]
```diff
@@ -80,6 +80,12 @@ This command prepares a Provider JSON metadata definition that can be used to ge
         'When set to true command will print the indexed documentation overview. This is useful for debugging.',
       default: false,
     }),
+    force: oclifFlags.boolean({
+      required: false,
+      description:
+        'When set to true command will overwrite existing Provider JSON metadata.',
```
Comment (kysely) [inline src/commands/prepare.ts:86]:
```suggestion
        'When set to `true`, the command will overwrite any existing Provider JSON metadata.',
```

**Review by kysely — COMMENTED** [review]
Looks fine.

One question though — when initially discussing this on a call, didn't we agree that CLI will automatically set/retry the force flag is the provider already exists? Or what is with something else?

**Comment by Jakub-Vacek** [comment]
> Looks fine.
> 
> One question though — when initially discussing this on a call, didn't we agree that CLI will automatically set/retry the force flag is the provider already exists? Or what is with something else?

Yes we did and decided that it is [optional](https://www.notion.so/superface/EDGAR-f318fcb914f347d888c4855b4794f9ab?pvs=4#669ac74c2774440ca5fae10bed7df88f). I wanted to focus more on AI related tasks (provider gen, analysis) but if you think we should implement it I will do it :)

**Comment by kysely** [comment]
> > Looks fine.
> > One question though — when initially discussing this on a call, didn't we agree that CLI will automatically set/retry the force flag is the provider already exists? Or what is with something else?
> 
> Yes we did and decided that it is [optional](https://www.notion.so/superface/EDGAR-f318fcb914f347d888c4855b4794f9ab?pvs=4#669ac74c2774440ca5fae10bed7df88f). I wanted to focus more on AI related tasks (provider gen, analysis) but if you think we should implement it I will do it :)

Hmm, I thought this optionality means “if we have time to do it, let's do it”, but this feature means:
- if provider already exists, ask user if they want to rewrite it, then retry the request with force flag

But probably let's not do it now if you don't feel like it

**Review by kysely — APPROVED** [review]
Good 👍

**Pull request merged**